add new resource OptionGroup

kprahulraj opened the pull request:
Issue [#2065](https://github.com/aws-controllers-k8s/community/issues/2065)

Description of changes:

The Amazon RDS [OptionGroup](https://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_CreateOptionGroup.html) resource represents the RDS Aurora Serverless API's command-line/configuration options for DB clusters.

- [x] generator config file to un-ignore OptionGroup resource
- [x] generate RDS service controller with support for the OptionGroup resource
- [x] e2e python test for CRUD operations for a OptionGroup


By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

kprahulraj committed:
add new resource OptionGroup

kprahulraj committed:
add new resource OptionGroup

michaelhtm commented on apis/v1alpha1/option_group.go:
```diff
@@ -0,0 +1,144 @@
+	MajorEngineVersion *string `json:"majorEngineVersion"`
+	// The description of the option group.
+	// +kubebuilder:validation:Required
+	OptionGroupDescription *string `json:"optionGroupDescription"`
```
can we rename this to `Description`?

knottnt replied:
@kprahulraj here's a [link](https://aws-controllers-k8s.github.io/community/docs/contributor-docs/code-generator-config/#renames-renaming-things) to the contributor docs section covering renames.

michaelhtm commented on apis/v1alpha1/option_group.go:
```diff
@@ -0,0 +1,144 @@
+	//
+	// Example: myoptiongroup
+	// +kubebuilder:validation:Required
+	OptionGroupName *string `json:"optionGroupName"`
```
same here..we can rename this to `Name`

kprahulraj replied:
@michaelhtm  fixed

michaelhtm reviewed (commented):
Thank you @kprahulraj 
left a few comments below

knottnt reviewed (commented):
Looking at the API Reference for [ModifyOptionGroup](https://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_ModifyOptionGroup.html) I can see that it provides request parameters for configuring Options within the OptionGroup. I think we likely need to update this PR so that users can configure those values.

michaelhtm committed:
feat: support configuring and updating OptionGroup options

Add Spec.Options (from ModifyOptionGroup.OptionsToInclude) with a
create-then-modify flow and a custom update that computes the
OptionsToInclude/OptionsToRemove sets. Observed options surface in
Status.OptionStatuses. Mark Description, EngineName and
MajorEngineVersion immutable to avoid an infinite reconcile loop on
fields ModifyOptionGroup cannot change.
